internal/auth/oidc/repository_auth_method_operational_state.go

```diff
@@ -0,0 +1,98 @@
+package oidc
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/boundary/internal/db"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/kms"
+	"github.com/hashicorp/boundary/internal/oplog"
+)
+
+// MakeInactive will transision an OIDC auth method from either the
+// ActivePrivateState or the ActivePublicState to the InactiveState.
+// No options are supported.
+func (r *Repository) MakeInactive(ctx context.Context, authMethodId string, _ ...Option) error {
```

Comment:
Since this is modifying the state of the auth method, I think it should take the version number, right?

Also in other methods which change a single field the pattern has been to return the full resource with the updated version number.  See SetPassword for password accounts as an example or Add|Set|Remove Group members for groups, or CancelSession in the session repo.

Reply:
I actually didn't think any of your assertions would be true, given that it's not an "updatable" field but rather it's a calculated field.  With that said, I have no strong preference and I'm trying to stay focused on moving forward.